src/charm.py

```diff
@@ -114,6 +115,19 @@ def _get_relation_data_or_config(self, field, fallback):
 
         return fallback
 
+    @property
+    def _additional_hostnames(self) -> List[str]:
+        """Return a list with additional hostnames.
+
+        Returns:
+            The additional hostnames set by configuration already split by comma.
+        """
+        return [
```

Comment:
Thanks for pointing that out but since one of the use cases is to check if a hostname is in the list (current hostname + all additional hostnames), an iterator wouldn't be a good fit.

Reply:
That is on the user side, it doesn't stop you from returning an iterator and the caller passing it to the list constructor. By returning a list, the use of a list is forced rather than up to the caller. The caller could be just interested in the first one or just need to process it

Reply:
Changed :)